contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,573 @@
+
+    /**
+     * Copies the character contents of a string to a local C string buffer after
+     * converting the characters to a given encoding.
+     *
+     * @param theString
+     *            The string whose contents you wish to access.
+     * @param bufferToFill
+     *            The C string buffer into which to copy the string. On return, the
+     *            buffer contains the converted characters. If there is an error in
+     *            conversion, the buffer contains only partial results.
+     *            <p>
+     *            The buffer must be large enough to contain the converted
+     *            characters and a NUL terminator.
+     * @param bufferSize
+     *            The length of {@code buffer} in bytes.
+     * @param encoding
+     *            The string encoding to which the character contents of
+     *            {@code theString} should be converted. The encoding must specify
+     *            an 8-bit encoding.
+     * @return {@code true} upon success or {@code false} if the conversion fails or
+     *         the provided buffer is too small.
+     */
+    boolean CFStringGetCString(CFTypeRef theString, Pointer bufferToFill, long bufferSize, int encoding);
+
+    /**
+     * Returns the value of a {@code CFBoolean} object.
+     *
+     * @param bool
+     *            The boolean to examine.
+     * @return The value of {@code bool}.
+     */
+    boolean CFBooleanGetValue(CFTypeRef bool);
+
+    /**
+     * Retrieves a value at a given index.
+     *
+     * @param theArray
+     *            The array to examine.
+     * @param idx
+     *            The index of the value to retrieve. If the index is outside the
+     *            index space of {@code theArray} (0 to N-1 inclusive (where N is
+     *            the count of {@code theArray})), the behavior is undefined.
+     * @return The value at the {@code idx} index in {@code theArray}).
+     */
+    CFTypeRef CFArrayGetValueAtIndex(CFArrayRef theArray, int idx);
+
+    /**
+     * Returns the type used by a {@code CFNumber} object to store its value.
+     *
+     * @param number
+     *            The {@code CFNumber} object to examine.
+     * @return A constant that indicates the data type of the value contained in
+     *         number. See {@link CFNumberType} for a list of possible values.
+     */
+    int CFNumberGetType(CFTypeRef number);
+
+    /**
+     * Obtains the value of a {@code CFNumber} object cast to a specified type.
+     *
+     * @param number
+     *            The {@code CFNumber} object to examine.
+     * @param theType
+     *            A constant that specifies the data type to return. See
+     *            {@link CFNumberType} for a list of possible values.
+     * @param On
+     *            return, contains the value of {@code number}.
+     */
+    void CFNumberGetValue(CFTypeRef number, int theType, ByReference valuePtr);
```

Comment:
- This is not a void method: https://developer.apple.com/documentation/corefoundation/1543114-cfnumbergetvalue
- the first parameter should be a CFNumberGetValue shouldn't it?

Reply:
Assuming you meant `CFNumberRef`.  This ended up being a shortcut I took to simplify code at the cost of type safety.  The issue here is that the first argument will be obtained from a call to `CFDictionaryGetValue()` which returns a pointer without the type specified.   In order to more narrowly cast it, we would have to create a `CFNumberRef` and then use `setPointer()` on it, using the returned value's pointer (still ignoring type safety but at least immediately adjacent to the call in the code). Should I revert to that approach?

Reply:
I've narrowed the type conversion on all these similar ones, using the `PointerType()` constructor to cast the returned values from elsewhere.  A few extra lines of code but better matches the method signatures.